SocketRocket/SocketRocket/Internal/Proxy/ARTSRProxyConnect.m

```diff
@@ -287,7 +287,28 @@ - (void)_runPACScript:(NSString *)script withProxySettings:(NSDictionary *)proxy
         NSString *proxyType = settings[(NSString *)kCFProxyTypeKey];
         [self _readProxySettingWithType:proxyType settings:settings];
     }
-    [self _openConnection];
+    [self openConnection];
+}
+
+- (void)openConnection
+{
+    if ([NSRunLoop ARTSR_networkRunLoop] != nil) {
+        [self _openConnection];
+    }
+    else {
+        if ([NSThread isMainThread]) {
```

Comment:
Could you add a comment explaining why we have this conditional logic that checks whether we're running on the main thread?

Reply:
That's because later I call `dispatch_async(dispatch_get_main_queue()...`. On the other side of this `if`, there is some background thread in which I can call `[NSRunLoop ARTSR_waitForNetworkRunLoop]` without blocking the main thread. This is for cases, where user decided to call `ARTRealtime` methods from not the main thread.

Reply:
That's what I imagined - but it would be useful to have this as a comment in the code.

Reply:
https://github.com/ably/ably-cocoa/pull/1538/commits/23f2ca1e867bf4523c06ce72487cccccdbec58ea
https://github.com/ably/ably-cocoa/pull/1538/commits/de192b1f6d70315fb92e2e1eb9e9f0b32bc64572